### PR TITLE
add StructField Metadata support

### PR DESCRIPTION
**Pull request by lesbroot**

### CODE REVIEW

**Comment by kmate** [comment]
Based on #18, AFAIK.

**Comment by szdavid92** [comment]
why did you create a separate PR for this?

**Comment by kmate** [comment]
It is a different feature.

File: core/src/main/scala/ste/encoder.scala [hunk 1]
```diff
@@ -23,23 +23,31 @@ package ste
 
 import org.apache.spark.sql.types._
 import shapeless._
+import shapeless.ops.hlist._
 import shapeless.labelled.FieldType
+import scala.annotation.StaticAnnotation
 
 import scala.collection.generic.IsTraversableOnce
 
+final class MetaData(val metadata: Metadata) extends StaticAnnotation
```
Comment (kmate) [inline core/src/main/scala/ste/encoder.scala]: Can we rename this to @Meta just to make usage shorter?

Comment (lesbroot) [reply core/src/main/scala/ste/encoder.scala]: done

File: core/src/main/scala/ste/encoder.scala [hunk 2]
```diff
@@ -23,10 +23,14 @@ package ste
 
 import org.apache.spark.sql.types._
 import shapeless._
+import shapeless.ops.hlist._
 import shapeless.labelled.FieldType
+import scala.annotation.StaticAnnotation
 
 import scala.collection.generic.IsTraversableOnce
 
+final class Meta(val metadata: Metadata) extends StaticAnnotation
```
Comment (BenFradet) [inline core/src/main/scala/ste/encoder.scala]: Do we need the `val` here?

Comment (kmate) [reply core/src/main/scala/ste/encoder.scala]: I think yes, because it is not a case class and we need to access this field later to extract the value.

**Pull request merged**